src/market/market_utils.cairo

```diff
@@ -2234,145 +2136,993 @@ fn get_virtual_inventory_for_swaps(
+    let ui_fee_factor: u128 = data_store.get_u128(keys::ui_fee_factor_key(account));
+
+    if ui_fee_factor < max_ui_fee_factor {
+        return ui_fee_factor;
+    } else {
+        return max_ui_fee_factor;
+    }
+}
+
+fn set_ui_fee_factor(
+    data_store: IDataStoreDispatcher,
+    event_emitter: IEventEmitterDispatcher,
+    account: ContractAddress,
+    ui_fee_factor: u128
+) {
+    let max_ui_fee_factor: u128 = data_store.get_u128(keys::max_ui_fee_factor());
+
+    if (ui_fee_factor <= max_ui_fee_factor) {
+        MarketError::UI_FEE_FACTOR_EXCEEDED(ui_fee_factor, max_ui_fee_factor);
+    }
+
+    data_store.set_u128(keys::ui_fee_factor_key(account), ui_fee_factor);
+
+    event_emitter.emit_ui_fee_factor_updated(account, ui_fee_factor);
+}
+
+fn validate_market_token_balance_array(data_store: IDataStoreDispatcher, markets: Array<Market>) {
+    let length: u32 = markets.len();
+    let mut i: u32 = 0;
+    loop {
+        if i == length {
+            break;
+        }
+        validate_market_token_balance_check(data_store, *markets.at(i));
+        i += 1;
+    };
+}
+
+fn validate_market_address_token_balance(
+    data_store: IDataStoreDispatcher, market_add: ContractAddress
+) {
+    let market: Market = get_enabled_market(data_store, market_add);
+    validate_market_token_balance_check(data_store, market);
+}
+
+fn validate_market_token_balance_check(data_store: IDataStoreDispatcher, market: Market) {
+    validate_market_token_balance_with_token(data_store, market, market.long_token);
+
+    if (market.long_token == market.short_token) {
+        return;
+    }
+    validate_market_token_balance_with_token(data_store, market, market.short_token);
+}
+
+fn validate_market_token_balance_with_token(
+    data_store: IDataStoreDispatcher, market: Market, token: ContractAddress
+) {
+    assert(
+        market.market_token.is_non_zero() && token.is_non_zero(),
+        MarketError::EMPTY_ADDRESS_IN_MARKET_TOKEN_BALANCE_VALIDATION
+    );
+    let balance_token: u256 = IERC20Dispatcher { contract_address: token }
+        .balance_of(market.market_token);
+    let balance: u128 = get_u256_low(balance_token);
+    let expected_min_balance: u128 = get_expected_min_token_balance(data_store, market, token);
+
+    assert(balance >= expected_min_balance, MarketError::INVALID_MARKET_TOKEN_BALANCE);
+
+    // funding fees can be claimed even if the collateral for positions that should pay funding fees
+    // hasn't been reduced yet
+    // due to that, funding fees and collateral is excluded from the expectedMinBalance calculation
+    // and validated separately
+
+    // use 1 for the getCollateralSum divisor since getCollateralSum does not sum over both the
+    // longToken and shortToken
+    let mut collateral_amount: u128 = get_collateral_sum(data_store, market, token, true, 1);
+    collateral_amount += get_collateral_sum(data_store, market, token, false, 1);
```

Comment:
You should fix the function param for market to take an address and use `market.market_token`